### PR TITLE
Adlive bid adapter

### PR DESCRIPTION
**Pull request by mifanich**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: 'adlive',
  params: {
    hashes: ['1e100887dd614b0909bf6c49ba7f69fdd1360437']
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
https://github.com/prebid/prebid.github.io/pull/970

### CODE REVIEW

File: modules/adliveBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      });
+
+      return bidResponses;
+    } catch (err) {
+      utils.logError(err);
+      return [];
+    }
+  },
+
+  getUserSyncs: function (syncOptions, serverResponses, gdprConsent) {
```
Comment (jsnellbaker) [inline modules/adliveBidAdapter.js]: If you don't have any userSync pixels you want to fire, you can just remove the entire `getUserSyncs` property from your adapter.  It's not required to define it.

Comment (mifanich) [reply modules/adliveBidAdapter.js]: If I do not define this function, tests do not pass.

Comment (jsnellbaker) [reply modules/adliveBidAdapter.js]: Can you clarify what tests do not pass?  I tried to disabled that part of your adapter code in the branch copy I checked out and ran`gulp test`.  There were no unit tests errors from what I saw.

Comment (jsnellbaker) [reply modules/adliveBidAdapter.js]: @mifanich I ran the browserstack tests (which is what circleci runs for us automatically) locally with the `getUserSyncs` code disabled and I didn't see any test failures.

It's possible there was some other issue that happened in the circleci tests you saw that failed that were unrelated to your adapter's code.  Can you try to make the change again and push it in this PR so we can retest it?

**Comment by jsnellbaker** [comment]
@mifanich Looks like the circleci test without the userSync code ran fine this time.  Please let me know about the delivery/response issue when you get the chance.  Thanks!

**Comment by jsnellbaker** [comment]
@mifanich I tried to test the adapter again with the params you updated in the last commit.  However I'm still seeing an error when the request is going out.

`OPTIONS https://api.publishers.adlive.io/get?pbjs=1 net::ERR_CONNECTION_REFUSED`

I have attached screenshots of the request as seen in the Console and Network tabs.  Can you take a look?

adlive_request_console_tab
adlive_request_network_tab

**Comment by mifanich** [comment]
I have changed test params and improve some logic.
gulp test works fine but circleci tests do not passed.

**Comment by jsnellbaker** [comment]
@mifanich Thanks for the update, I'll check again.  The circleci failed for an unrelated reason; I restarted the build and it completed fine.

**Comment by jsnellbaker** [comment]
@mifanich I retested the adapter and I was seeing a response coming back from the server.  However the bid wasn't being accepted as per the logic in the `interpretResponse` function.

The response contained a `1` in the `bidResponse.is_passback`, which caused the if statement check at https://github.com/prebid/Prebid.js/pull/3109/files#diff-841514cf584a84bb2bb332698c4b54feR46 to fail to pass.  Is this expected?  

Below is a copy of the response object I saw:
```
[{"hash":"1e100887dd614b0909bf6c49ba7f69fdd1360437","content":"<script async src=\"//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js\"></script> <!-- lacuisinedebernard_300x250 --> <ins class=\"adsbygoogle\"      style=\"display:inline-block;width:300px;height:250px\"      data-ad-client=\"ca-pub-7357742692072135\"      data-ad-slot=\"9217654942\"></ins> <script> (adsbygoogle = window.adsbygoogle || []).push({}); </script>","size":[300,250],"is_passback":1,"bids":[{"adapter":"pulsepoint","nobid":true},{"adapter":"districtm","nobid":true},{"adapter":"appnexus","nobid":true}]}]
```

**Comment by mifanich** [comment]
> @mifanich I retested the adapter and I was seeing a response coming back from the server. However the bid wasn't being accepted as per the logic in the `interpretResponse` function.
> 
> The response contained a `1` in the `bidResponse.is_passback`, which caused the if statement check at https://github.com/prebid/Prebid.js/pull/3109/files#diff-841514cf584a84bb2bb332698c4b54feR46 to fail to pass. Is this expected?
> 
> Below is a copy of the response object I saw:
> 
> ```
> [{"hash":"1e100887dd614b0909bf6c49ba7f69fdd1360437","content":"<script async src=\"//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js\"></script>  <ins class=\"adsbygoogle\"      style=\"display:inline-block;width:300px;height:250px\"      data-ad-client=\"ca-pub-7357742692072135\"      data-ad-slot=\"9217654942\"></ins> <script> (adsbygoogle = window.adsbygoogle || []).push({}); </script>","size":[300,250],"is_passback":1,"bids":[{"adapter":"pulsepoint","nobid":true},{"adapter":"districtm","nobid":true},{"adapter":"appnexus","nobid":true}]}]
> ```

Yes, you are right! If response contains is_passback=1 it means there is no bids. Need to reload page several times because this is production endpoint. Bids may not present.

**Comment by mifanich** [comment]
@jsnellbaker please check updates

**Comment by jsnellbaker** [comment]
@mifanich Is there some way you can setup a dedicated test?  I've been trying to get something to return off/on for a while now and I haven't seen it once return an actual bid.

**Comment by jsnellbaker** [comment]
@mifanich Thanks, let me know when it's ready and if I need to use any different params.

**Comment by mifanich** [comment]
@jsnellbaker Fixed.
Please use https://api.publishers.adlive.io/get?pbjs=1&pb_test=1 as POST URL

**Comment by jsnellbaker** [comment]
That test param did work and I saw the test ad; LGTM

**Pull request merged**